### PR TITLE
Feature/context response

### PR DESCRIPTION
**Pull request by froggomad**
# Members can respond to context questions

## Changes
- Add ContextReponse
- Add Thread

## Checklist
`choose one minimum`
- [x] created unit test
- [x] updated unit test
- [ ] created UI test
- [ ] updated UI test
- [ ] bug fix/other

`required`
- [x] tested locally
- [x] updated the docs (https://github.com/Lambda-School-Labs/Labs26-Apollo-iOS-TeamA/blob/master/.github/documentation_standards.md)

### CODE REVIEW

File: labs-ios-starter/Surveys/View Controllers/TopicViewController.swift [hunk 1]
```diff
@@ -110,6 +113,8 @@ class TopicViewController: LoginViewController, NSFetchedResultsControllerDelega
                         self.spinner.stopAnimating()
                         topicsCollectionView.reloadData()
                     } catch {
+                        self.spinner.stopAnimating()
+                        topicsCollectionView.reloadData()
```
Comment (Shawn-James) [inline labs-ios-starter/Surveys/View Controllers/TopicViewController.swift:117]: good catch. pun intended

File: labs-ios-starterTests/CoreDataTests.swift [hunk 2]
```diff
@@ -14,48 +14,59 @@ class CoreDataTests: XCTestCase {
...
+
+        let response = ContextResponse(id: 999, questionId: contextQuestion.id, response: "IDK, why don't we go to HR and find out", respondedBy: member, contextQuestion: contextQuestion)
+
+        contextQuestion.addToResponse(response)
         do {
             try CoreDataManager.shared.saveContext()
         } catch {
             XCTFail("error saving CoreData: \(error)")
         }
+
```
Comment (Shawn-James) [inline labs-ios-starterTests/CoreDataTests.swift:40]: nice TDD

**Review by Shawn-James — APPROVED** [review]
Looks good. Makes me wish I spent more time writing tests

File: labs-ios-starter/Surveys/View Controllers/TopicViewController.swift (hunk 1 above)
Comment (froggomad) [reply labs-ios-starter/Surveys/View Controllers/TopicViewController.swift:117]: thx... we could probably use `defer { }` here, but that always worries me lol

**Pull request merged**